### PR TITLE
[chore] Cleanup GitLab source errors

### PR DESCRIPTION
**Pull request by mcastorina**
### Description:
Cleanup error handling and creation in GitLab source.

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/sources/gitlab/gitlab.go [hunk 1]
```diff
@@ -408,15 +409,16 @@ func (s *Source) getReposFromGitlab(ctx context.Context, apiClient *gitlab.Clien
 		// Ensure the urls are valid before adding them to the repo list.
 		_, err := url.Parse(prj.HTTPURLToRepo)
 		if err != nil {
-			fmt.Printf("could not parse url given by project: %s", prj.HTTPURLToRepo)
+			ctx.Logger().Error(err, "could not parse url given by project", "project", prj.HTTPURLToRepo)
+			continue
 		}
 		repos = append(repos, prj.HTTPURLToRepo)
```
Comment (mcastorina) [inline pkg/sources/gitlab/gitlab.go:415]: Control flow change here (added the `continue`)

File: pkg/sources/gitlab/gitlab.go [hunk 2]
```diff
@@ -463,20 +465,20 @@ func (s *Source) scanRepos(ctx context.Context, chunksChan chan *sources.Chunk)
 				}
 				path, repo, err = git.CloneRepoUsingToken(ctx, s.token, repoURL, user)
 			}
-			defer os.RemoveAll(path)
 			if err != nil {
 				scanErrs.Add(err)
 				return nil
 			}
+			defer os.RemoveAll(path)
```
Comment (mcastorina) [inline pkg/sources/gitlab/gitlab.go:472]: Moved this `defer` to after the error check.

**Review by rosecodym — APPROVED** [review]
what prompted this?

(also, any idea why it's re-scanning the whole repo for secrets?)

**Comment by mcastorina** [comment]
> what prompted this?

I started adding source unit support for gitlab and decided to clean it up (separately) while I was there.

> (also, any idea why it's re-scanning the whole repo for secrets?)

Nope, though I think it's been happening for awhile.. it's tracked in `THOG-1640`

File: pkg/sources/gitlab/gitlab.go [hunk 3]
```diff
@@ -108,7 +109,7 @@ func (s *Source) Init(_ context.Context, name string, jobId sources.JobID, sourc
 		// We may need the password as a token if the user is using an access_token with basic auth.
 		s.token = cred.BasicAuth.Password
 	default:
-		return errors.Errorf("Invalid configuration given for source. Name: %s, Type: %s", name, s.Type())
+		return fmt.Errorf("invalid auth method")
```
[the remark this answers is not in the document]
Comment (mcastorina) [reply pkg/sources/gitlab/gitlab.go:112]: hmm.. I'm not entirely sure. I don't think we can actually trigger this error 🤔

File: pkg/sources/gitlab/gitlab.go [hunk 4]
```diff
@@ -53,9 +52,11 @@ type Source struct {
 }
 
 // Ensure the Source satisfies the interfaces at compile time.
-var _ sources.Source = (*Source)(nil)
-var _ sources.SourceUnitUnmarshaller = (*Source)(nil)
-var _ sources.Validator = (*Source)(nil)
+var _ interface {
```
Comment (ahrav) [inline pkg/sources/gitlab/gitlab.go:55]: Did we decide to use this version of the compile check? I know there was some convo out of band around errors not being as clear, not sure if that was a completely separate convo or related to this?

**Review by ahrav — APPROVED** [review]

File: pkg/sources/gitlab/gitlab.go (hunk 4 above)
Comment (rosecodym) [reply pkg/sources/gitlab/gitlab.go:55]: that was me asking your question internally 😄

File: pkg/sources/gitlab/gitlab.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rosecodym) [reply pkg/sources/gitlab/gitlab.go:112]: i'm sure enterprise can with a malformed config file. might be worth double-checking.

File: pkg/sources/gitlab/gitlab.go (hunk 4 above)
Comment (mcastorina) [reply pkg/sources/gitlab/gitlab.go:55]: This PR probably prompted the convo - I'll change it back

**Pull request merged**